doc/hooks/patch-activation-hooks.section.md

```diff
@@ -0,0 +1,49 @@
+
+# `patchActivation` hooks {#sec-patchActivationHooks}
+
+These hooks provide shell-specific utilities (with the same name as the hook) to patch shell scripts meant to be sourced by software users.
+
+The typical usage is to patch the "activation scripts" inside `$out/bin`;
```

Comment:
@ShamrockLee sorry for the long waiting time. I was kind of procrastinating taking another look at the documentation. This PR always produced a weird feeling, because I could get rid of the impression of not fully understanding what the new feature does.

Took a look again today, and now I see the problem: You call these wrappers "activation scripts", and that confused the hell out of me. [`activationScript`](https://nixos.org/manual/nixos/stable/#sec-activation-script) is a completely unrelated concept in NixOS (and in itself misleading, because too low-level for most users to care about).

Since Nixpkgs users with high probability are also NixOS users, I expect the "activation" part of the naming to produce lots of confusion for other readers, too.

Please correct me if I'm wrong here. I understand your patch tools are for wrapper scripts around binaries in packages. They should not be called "activation scripts", for the above reason. If you confirm that understand this correctly, I reluctantly veto this PR until we find a more suitable name. Sorry for the additional work that would entail for you.

Proposal: `patchWrapper` hooks.

Reply:
Thanks a lot for pointing out the problem! I don't even notice that the naming conflict to that widely-used infrastructure.

The name I originally choose was `patchSourcetimePath` hooks, as they provided a temporary `PATH` prefix to the `source`-able shell script, but later replaced it to make it shorter.

`patchWrapper` sounds good, but I don't expect people to use it on script wrappers generated by `makeWrapper`. How about `patchRcPath`? Those scripts are usually sourced before running the main program.

Reply:
Agreed that "wrapper" brings in a context that does not apply here.

`patchRcPath` is good, because it's technically correct and concise. It would be really valuable to add a brief note in the manual section (and link to the source) explaining what `rc` means here, otherwise the name will be unhelpful to everyone except the most greybearded Unixoids.

For example: https://kb.iu.edu/d/abnd via https://unix.stackexchange.com/questions/111611/what-does-the-rc-stand-for-in-etc-rc-d

Reply:
Added.